### ISSUE TITLE
Custom types can't have authorization modes added to them, meaning if you have a custom mutation and multiple auth modes, response types won't be accessible

### ISSUE THREAD

**Issue by aherschel** [aws-amplify/amplify-backend#1359]
### Environment information

```plain text
Relevant deps:

"@aws-amplify/data-schema": "^0.16.2",
"@aws-amplify/graphql-api-construct": "^1.8.1",
```
```


### Description

This issue may be better routed to the graphql-cdk repo, but dropping here as I think it relates to Gen2 release.

```typescript
// schema.ts
import { a, type ClientSchema } from '@aws-amplify/data-schema';

export const schema = a.schema({
  EchoResponse: a.customType({
    content: a.string().required(),
    executionDuration: a.float().required(),
  }),
  echo: a
    .query()
    .arguments({ content: a.string().required() })
    .returns(a.ref('EchoResponse'))
    .authorization(allow => [
      allow.publicApiKey(),
    ])
    .handler(a.handler.function('echo'))
});

export type Schema = ClientSchema<typeof schema>;
```

```typescript
// app.ts

...
    const echo = createNodeJsFunction('EchoFn', 'echo.ts'); // generates a lambda fn, details shouldn't matter here

    const api = new AmplifyGraphqlApi(graphqlStack, 'Api', {
      definition: AmplifyGraphqlDefinition.fromString(schema.transform().schema),
      authorizationModes: {
        defaultAuthorizationMode: 'AWS_IAM',
        apiKeyConfig: { expires: Duration.days(30) },
        userPoolConfig: { userPool: auth.resources.userPool },
        iamConfig: {
          identityPoolId: auth.resources.cfnResources.cfnIdentityPool.ref,
          authenticatedUserRole: auth.resources.authenticatedUserIamRole,
          unauthenticatedUserRole: auth.resources.unauthenticatedUserIamRole,
        },
      },
      functionNameMap: {
        echo,
      },
    });
...
```

Basically, the authmode configured on the ref'd type isn't having the relevant auth modes enabled on the response types. I had to manually update the schema post-hoc to add the @aws_api_key annotation to the schema after deploying to test.

**Comment by ykethan** [comment]
Hey👋 thanks for raising this! I'm going to transfer this over to our API repository for better assistance 🙂
